File: app/src/main/java/com/raival/fileexplorer/tabs/file/model/Task.java
package com.raival.fileexplorer.tabs.file.model;

import java.io.File;
import java.util.ArrayList;

public abstract class Task {
    public interface OnUpdateListener{
        void onUpdate(String progress);
    }
    public interface OnFinishListener{
        void onFinish(String result);
    }

    public abstract String getName();
    public abstract String getDetails();
    public abstract boolean isValid();
    public abstract void setActiveDirectory(File directory);
    public abstract void start(OnUpdateListener onUpdate, OnFinishListener onFinish);
}

File: app/src/main/java/com/raival/fileexplorer/tab/file/model/Task.kt
package com.raival.fileexplorer.tab.file.model

import java.io.File

abstract class Task {
    abstract val name: String?
    abstract val details: String?
    abstract val isValid: Boolean

    abstract fun setActiveDirectory(directory: File)
    abstract fun start(onUpdate: OnUpdateListener, onFinish: OnFinishListener)

    interface OnUpdateListener {
        fun onUpdate(progress: String)
    }

    interface OnFinishListener {
        fun onFinish(result: String)
    }
}